src/ImageSharp/Processing/Processors/Effects/OilPaintingProcessor{TPixel}.cs

```diff
@@ -52,82 +54,107 @@ protected override void OnFrameApply(ImageFrame<TPixel> source)
 
             int radius = brushSize >> 1;
             int levels = this.definition.Levels;
+            int rowWidth = source.Width;
+            int rectangleWidth = this.SourceRectangle.Width;
 
-            using (Buffer2D<TPixel> targetPixels = this.Configuration.MemoryAllocator.Allocate2D<TPixel>(source.Size()))
-            {
-                source.CopyTo(targetPixels);
+            Configuration configuration = this.Configuration;
 
-                var workingRect = Rectangle.FromLTRB(startX, startY, endX, endY);
-                ParallelHelper.IterateRows(
-                    workingRect,
-                    this.Configuration,
-                    rows =>
-                        {
-                            for (int y = rows.Min; y < rows.Max; y++)
-                            {
-                                Span<TPixel> sourceRow = source.GetPixelRowSpan(y);
-                                Span<TPixel> targetRow = targetPixels.GetRowSpan(y);
+            using Buffer2D<TPixel> targetPixels = this.Configuration.MemoryAllocator.Allocate2D<TPixel>(source.Size());
+            source.CopyTo(targetPixels);
 
-                                for (int x = startX; x < endX; x++)
-                                {
-                                    int maxIntensity = 0;
-                                    int maxIndex = 0;
+            var workingRect = Rectangle.FromLTRB(startX, startY, endX, endY);
+            ParallelHelper.IterateRows(
```

Comment:
`IterateRowsWithTempBuffer`

Reply:
I'm not using that on purpose, since I need two buffers anyway. Plus, the buffer I need needs to be as wide as the entire row, not just the target rectangle area, so using that helper method is not possible in this case.
Should I add a detailing comment there to explain the rationale?

Reply:
Ah ok, sorry. I didn't realize it was only the size of the working rectangle.

Reply:
No worries, that was actually a very good point and I had the same doubt at first. You were absolutely right to ask me to clarify that, and I think having a more detailed comment in there will definitely come in handy when (if) someone else will look back at this code again in a few months 👍